Sources/BraveWallet/Crypto/Stores/NFTStore.swift

```diff
@@ -235,100 +270,176 @@ public class NFTStore: ObservableObject, WalletObserverStore {
-      userNFTs = buildAssetViewModels(allUserAssets: allNetworkAssets)
-        .optionallyFilterUnownedNFTs(
-          isHidingUnownedNFTs: filters.isHidingUnownedNFTs,
-          selectedAccounts: selectedAccounts
-        )
+      userNFTGroups = buildNFTGroupModels(
+        groupBy: filters.groupBy,
+        allUserAssets: allNetworkAssets,
+        selectedAccounts: selectedAccounts,
+        selectedNetworks: selectedNetworks
+      )
     }
   }
   
   func updateNFTMetadataCache(for token: BraveWallet.BlockchainToken, metadata: NFTMetadata) {
     metadataCache[token.id] = metadata
-    if let index = userNFTs.firstIndex(where: { $0.token.id == token.id }),
-       var updatedViewModel = userNFTs[safe: index] {
-      updatedViewModel.nftMetadata = metadata
-      userNFTs[index] = updatedViewModel
+    var updatedGroups: [NFTGroupViewModel] = []
+    for group in userNFTGroups {
+      if let index = group.assets.firstIndex(where: { $0.token.id == token.id }) {
+        var newAssets = group.assets
+        newAssets[index].nftMetadata = metadata
+        updatedGroups.append(.init(groupType: group.groupType, assets: newAssets))
+      } else {
+        updatedGroups.append(group)
+      }
     }
+    userNFTGroups = updatedGroups
   }
   
   private func buildAssetViewModels(
+    for groupType: AssetGroupType,
     allUserAssets: [NetworkAssets]
   ) -> [NFTAssetViewModel] {
-    allUserAssets.flatMap { networkAssets in
-      networkAssets.tokens.compactMap { token in
-        guard token.isErc721 || token.isNft else { return nil }
-        return NFTAssetViewModel(
-          token: token,
-          network: networkAssets.network,
-          balanceForAccounts: nftBalancesCache[token.id] ?? [:],
-          nftMetadata: metadataCache[token.id]
-        )
+    let selectedAccounts = self.filters.accounts.filter(\.isSelected).map(\.model)
+    switch groupType {
+    case .none:
+      return allUserAssets.flatMap { networkAssets in
+        networkAssets.tokens.compactMap { token in
+          guard token.isErc721 || token.isNft else { return nil }
+          return NFTAssetViewModel(
+            groupType: groupType,
+            token: token,
+            network: networkAssets.network,
+            balanceForAccounts: nftBalancesCache[token.id] ?? [:],
+            nftMetadata: metadataCache[token.id]
+          )
+        }
+      }
+      .optionallyFilterUnownedNFTs(
+        isHidingUnownedNFTs: filters.isHidingUnownedNFTs,
+        selectedAccounts: selectedAccounts
+      )
+      .optionallySort(shouldSort: true) { first, second in
+        first.token.symbol < second.token.symbol
+      }
+    case let .network(network):
+      guard let networkAssets = allUserAssets
+        .first(where: { $0.network.chainId == network.chainId && $0.network.coin == network.coin }) else {
+        return []
       }
+      return networkAssets.tokens
+        .filter { $0.isErc721 || $0.isNft }
```

Comment:
I think we'd benefit from filtering non-NFTs out earlier in `update()` (or in shared function for fetching assets 😄); it would also improve performance of lookups on the 2 user asset arrays.

Additionally it would help avoid potential issues like https://github.com/brave/brave-ios/pull/8300/files